backend/src/login/authenticate.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+import jwt
+
+from src import context
+from src.login.user.role import Role
+from src.secret import CloudSecretManager
+
+
+@context.inject
+def has_valid_token(token, environment, cloud_secret_manager: CloudSecretManager = None) -> bool:
+    public_key = cloud_secret_manager.get_secret(f"document-extractor-{environment}-public-key")
+    try:
+        jwt.decode(token, public_key, algorithms=["RS512"])
+        return True
+    except jwt.PyJWTError as e:
+        exception_message = "Failed to authenticate token"
+        logging.error(exception_message)
+        logging.exception(e)
+        return False
```

Comment:
So... I think we shouldn't log the exception and we should log the exception message only at the warn level.  I think this because this isn't an unexpected exception.  It's expected sometimes.  I like to use warn level logging for that and I generally don't log the exception, especially if we not wrapping the exception, rethrowning the wrapped exception, and instead just returning `False` or something.

```suggestion
    except jwt.PyJWTError as e:
        exception_message = "Failed to authenticate token"
        logging.warn(exception_message)
        return False
```

Reply:
I changed it from `logging.warn` to `logging.warning` because after I ran pytest I got a deprecated method on `logging.warn' and it suggested the latter.

Reply:
@saquino0827, perfect.  Yeah, I couldn't remember what is the newer method.  Do you also want to get rid of the `logging.exception(e)` line too?

Reply:
Good catch, I thought I did get rid of it.